internal/functions/publishkeys/publish-keys.go

```diff
@@ -296,28 +298,42 @@ func extractDoS(request v1.PublishKeysRequestDevice) time.Time {
 	return time.Unix(soi*600, 0).Truncate(24 * time.Hour)
 }
 
-func updateCounters(ctx context.Context, client *realtimedb.Client, keysCount int) error {
+func updateCounters(ctx context.Context, client *realtimedb.Client, keysCount int, efgsEnabled bool) error {
 	logger := logging.FromContext(ctx).Named("publish-keys.updateCounters")
 
 	var date = utils.GetTimeNow().Format("20060102")
 
-	// update daily counter
-	if err := updateCounter(ctx, client, constants.DbPublisherCountersPrefix+date, keysCount); err != nil {
+	// update keys daily counter
+	if err := updateKeysCounter(ctx, client, constants.DbPublisherCountersPrefix+date, keysCount); err != nil {
 		logger.Warnf("Cannot increase publishers counter due to unknown error: %+v", err.Error())
 		return err
 	}
 
-	// update total counter
-	if err := updateCounter(ctx, client, constants.DbPublisherCountersPrefix+"total", keysCount); err != nil {
+	// update keys total counter
+	if err := updateKeysCounter(ctx, client, constants.DbPublisherCountersPrefix+"total", keysCount); err != nil {
 		logger.Warnf("Cannot increase publishers counter due to unknown error: %+v", err.Error())
 		return err
 	}
 
+	if efgsEnabled {
+		// update publishers daily counter
+		if err := updatePublishersCounter(ctx, client, constants.DbEfgsCountersPrefix+date); err != nil {
+			logger.Warnf("Cannot increase EFGS publishers counter due to unknown error: %+v", err.Error())
```

Comment:
This `err` is logged twice. Once here as `Warn` and the second time on the line [107](https://github.com/covid19cz/erouska-backend/blob/93924dee23972d3955b39db2ce7248a293cd9446/internal/functions/publishkeys/publish-keys.go#L107) as `Error`. Same in the whole function.

Reply:
You are very right :wink: